File: pinyougou/pinyougou-parent/pinyougou-user-interface/src/main/java/com/pinyougou/user/service/UserService.java
package com.pinyougou.user.service;

import com.pinyougou.model.User;

public interface UserService {

    /***
     * 增加User信息
     * @param user
     * @return
     */
    int add(User user);


    /**
     * 创建验证码
     * @param phone
     * @return
     */
    void createCode(String phone) throws Exception;

    /**
     * 根据用户名查询用户是否存在
     * @param username
     * @return
     */
    int getUserByUserName(String username);

    /**
     * 根据手机号查询用户是否存在
     * @param phone
     * @return
     */
    int getUserByPhone(String phone);

    /**
     * 判断验证码是否相同
     * @param phone
     * @param code
     * @return
     */
    Boolean checkCode(String phone, String code);

    /**
     * 根据用户名查询用户
     * @param name
     * @return
     */
    User findUserByUserName(String name);
}

File: pinyougou/pinyougou-parent/pinyougou-user-web/src/main/webapp/js/service/userService.js
/**
 * 创建service
 */
app.service("userService",function ($http) {

    /**
     *创建验证码
     * @param phone
     */
    this.createCode=function (phone) {
        return $http.get("/user/code.shtml?phone="+phone)
    }
    /**
     * 注册功能
     * @param entity
     */
    this.register=function (entity,code) {
        return $http.post("/user/add.shtml?code="+code,entity)
    }
});